File: tests/fixtures/vuln/Clean.cs
// A clean fixture — the enumerator must find NOTHING here. Not compiled/run.
using System.Data.SqlClient;

public class Clean
{
    static string ApiKey = System.Environment.GetEnvironmentVariable("API_KEY");

    void Lookup(SqlConnection c, string userId)
    {
        // parameterised query — the safe form
        var cmd = new SqlCommand("SELECT * FROM users WHERE id = @id", c);
        cmd.Parameters.AddWithValue("@id", userId);
    }
}

File: tests/fixtures/vuln/clean.rb
# A clean fixture — the enumerator must find NOTHING here. Not required/run.
require "digest"

API_KEY = ENV["API_KEY"] # reference, not a literal

def lookup(user_id)
  # parameterised query — the safe form
  User.where("id = ?", user_id)
end

def hash_pw(pw)
  Digest::SHA256.hexdigest(pw) # strong hash
end
